### DISCUSSION TITLE
Map.Lookup: invalid argument

### DISCUSSION THREAD

**Discussion by nashuiliang** [cilium/ebpf#1060] [Q&A]
Hi, We use ebpf@v0.9.0 in the production environment, and for the last 1 month or so, the errors of `Map.Lookup` are generated once a day. The simplified code is as follows:

- bpf.c

```c
struct softirq_lat {
        u64 silat[64];
};

struct {
        __uint(type, BPF_MAP_TYPE_ARRAY);
        __uint(key_size, sizeof(u32));
        __uint(value_size, sizeof(struct softirq_lat));
        __uint(max_entries, 1);
} softirq_lats SEC(".maps");
```

- bpf.go

```golang
func start() error {
    for {
        var siLat softirqLat
        if err := ebpfObjs.SoftirqLats.Lookup(int32(0), unsafe.Pointer(&siLat)); err != nil {
            return err
        }
        // do other things.
        // wait
        time.Sleep(1*time.Second)
    }
}
```

The error message is as follows:

```text
time='2023-06-12T08:19:10.845601752+08:00' level='warning' error='lookup: invalid argument'
```

The environment information is as follows:

```sh
$ go version
1.17
$ uname -r
4.18.0-193.6.3.el8_2.x86_64
```

We have carefully looked at the code of BPF_MAP_LOOKUP_ELEM in linux, but we have not found where this problem may occur. Do you have any ideas?

**Comment by lmb** [comment]
From your uname it seems like you're on centos, have you looked at the sources for that? It's usually heavily patched from upstream. Please note that we don't officially support non-kernel.org kernels so you'll have to figure this out on your own.
